### ISSUE TITLE
masking does not work in -a 3 but works in -a 6

### ISSUE THREAD

**Issue by komawoyo** [hashcat/hashcat#1617]
Referencing https://github.com/hashcat/hashcat/issues/1611
 
I believe my mask is correct. What am I doing wrong?
Correct Key: 3wafm3w4

![image](https://user-images.githubusercontent.com/18315233/42601804-821a0de2-852c-11e8-84a2-4225f3bbb9a9.png)

**Comment by philsmd** [comment]
Did you try using --opencl-devices (or short -d) to test the single GPU cracking? Why do you say it works with a single GPU ?
for instance you could use -d 1

btw you do not need to have the character "w" two times within the custom charset definition, it's enough to just specify it once like: -1 34afmw

**Comment by komawoyo** [comment]
Thank you for the suggestion, single GPU does not work also. fyi, these are all Vega 56

![image](https://user-images.githubusercontent.com/18315233/42630049-19c23a16-859b-11e8-8f4f-29e210e2056a.png)

**Comment by philsmd** [comment]
are you able to crack the example hccapx file from https://hashcat.net/wiki/example_hashes with password "hashcat!" with a mask attack -a 3 ?

btw: the title seems to be not correct if it is not a multi-GPU problem (we already excluded that). It seems that this is now a 100% duplicate of #1611 (I think we should close this and if you are 100% sure it is not somehow hardware/driver related we might rethink to reopen the older one).... on the other hand, I don't think that it is hashcat's fault because it works with your self-test and we also have no other similar reports (it could still be a very specific vega+WPA related problem)

**Comment by komawoyo** [comment]
I'm able to use -a 6 to crack the password so the masking seems to work at least for that option 

![image](https://user-images.githubusercontent.com/18315233/42662110-f9adaf5c-85f5-11e8-880d-25cac6e95646.png)

**Comment by philsmd** [comment]
Did the example hash work for you? Are you able to use -a 3 with other hash types?

The title is still missleading because it doesn't mention that this is (probably) a vega related problem with -m 2500 = WPA/WPA2. My guess is that your setup (hardware/driver) only have problem with WPA.
Did you test using other/older driver versions ? Did you try rocm on linux ?

**Comment by komawoyo** [comment]
I ran the examples within the folders and they worked fine. the moment I changed to -a 3 mode, it does not work. I do not think -m 2500 is the problem because its able to crack it just fine with -a 6 mode. I have tried other drivers. I have not tried on linux. Everything works on my Intel GPU.  not sure whats the issue with AMD and -a 3 mode when other attacks works perfectly fine - anyways,  I'll close this issue soon - it looks like no one has the same problem as me using AMD
